deepinv/training_utils.py

```diff
@@ -487,15 +485,9 @@ def validate(self, epoch):
 
                 with torch.no_grad():
 
-                    samples = self.get_samples(self.val_iterators, g)
-
-                    x, y, physics_cur = (
-                        samples["x"],
-                        samples["y"],
-                        samples["physics_cur"],
-                    )
+                    x, y, physics_cur = self.get_samples(self.val_iterators, g)
 
-                    x_net = self.model_inference(samples)
+                    x_net = self.model_inference(y=y, physics_cur=physics_cur)
```

Comment:
I like this. However, if the user wants to add another variable (e.g. `z`, or `labels`) to samples, they have to override `train_step`, `get_samples` and `model_inference`? I'm wondering if we could have a kwargs approach to not have to override `train_step`, but I don't want to go back to the dict approach, maybe like:
```
x, y, physics, kwargs = self.get_samples(...)
x_net = self.model_inference(y=y, physics_cur=physics, **kwargs)
``` 
... but this is quite ugly, so happy to have the user have to override `train_step` too if they need this functionality.